extension/INTEGRATING_HARDWARE_WALLET.MD

```diff
@@ -0,0 +1,90 @@
+   types: [@shared/constants/hardwareWallet.ts](other_file.md). In this file,
+   add the chosen variable name to the enum `WalletType`. The value of this enum
+   key will tell Freighter how to show your wallet's name in the Freighter UI.
+
+   This will tell Freighter to look for configurations for this new WalletType.
+   Because of that, you will notice right away that adding this to enum suddenly
+   causes Typescript errors across the app. We'll use these errors to identify
+   all the places we're required to add configuration for the new wallet. Code
+   example:
+
+   ```
+   export enum WalletType {
+   LEDGER = "Ledger",
+   YOUR_WALLET = "Your Wallet",
+   NONE = "",
+   }
+   ```
+
+2. Navigate to
+   [src/popup/helpers/hardwareConnect.ts](src/popup/helpers/hardwareConnect.ts).
+   Here we'll add configurations for the new wallet.
+
+3. First we'll add helper methods so Freighter knows how to interact the
+   hardware wallet's API. Start by adding a method for `createWalletConnection`.
+   You'll add a key to this object using the enum value you set above. The value
+   for this key will be an async function that runs any code needed to connect
+   Freighter to your wallet and then return a public key. This will be run
+   whenever a user adds your hardware wallet to Freighter for the first time.
+   Code example:
+
+```
+export const createWalletConnection: CreateWalletConnection = {
+  [WalletType.LEDGER]: async (bipPath = "") => {
+    const transport = await TransportWebUSB.request();
+    const ledgerApi = new LedgerApi(transport);
+    const response = await ledgerApi.getPublicKey(bipPath);
+
+    return response.publicKey;
+  },
+  [WalletType.YOUR_WALLET] async (bipPath = "") => {
+    // your code goes here
+    return "Gxxxxxxx";
+  }
+};
+```
+
+4. Similary, next we'll add a method to `getWalletPublicKey`. The method added
+   here will run any time Freighter needs to retrieve a public key from a
+   hardware wallet that has already been connected (for example, signing a
+   transaction).
+
+5. Next, we'll add a method to `hardwareSign`. This will be the code that runs
+   whenever Freighter tries to sign a transaction using your hardware wallet as
+   a signer. This code will likely make an API call that allows a user to sign
+   the transaction using prompts on their hardware wallet. Once complete, the
+   API will return a signature to Freighter. Freighter will then add that
+   signature to the transaction.
+
+6. With those methods set, Freighter now knows how to interact with the hardware
+   wallet. Now, we'll tell Freighter how to show information about your wallet.
+   Scroll down to `walletAssets`. Here we'll set the key `logo` with your
+   hardware wallet's logo. This will appear in a button when Freighter lists all
+   possible hardware wallets to connect with. Drop your assets into the
+   [src/popup/assets/](src/popup/assets/) folder. You can use almost any image
```

Comment:
Should we document or enforce specific dimensions or resolutions in order to create some consistency across assets from different providers?

Reply:
We do, actually. I list the recommended dimensions in the helper file

Reply:
oh I missed that, thanks.